presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestInlineSqlFunctions.java

```diff
@@ -0,0 +1,160 @@
+import com.facebook.presto.spi.function.SqlInvokedFunction;
+import com.facebook.presto.spi.plan.AggregationNode;
+import com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder;
+import com.facebook.presto.sql.planner.iterative.rule.test.RuleTester;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.NodeRef;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.facebook.presto.sql.tree.SymbolReference;
+import com.facebook.presto.testing.TestingSession;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.common.type.StandardTypes.INTEGER;
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.spi.function.RoutineCharacteristics.Determinism.DETERMINISTIC;
+import static com.facebook.presto.spi.function.RoutineCharacteristics.NullCallClause.RETURNS_NULL_ON_NULL_INPUT;
+import static com.facebook.presto.sql.analyzer.ExpressionAnalyzer.getExpressionTypes;
+import static com.facebook.presto.sql.planner.TypeProvider.viewOf;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.aggregation;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.expression;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.functionCall;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.values;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.assignment;
+import static org.testng.Assert.assertEquals;
+
+public class TestInlineSqlFunctions
+{
+    private static final SqlInvokedFunction SQL_FUNCTION_SQUARE = new SqlInvokedFunction(
+            QualifiedFunctionName.of(new CatalogSchemaName("unittest", "memory"), "square"),
+            ImmutableList.of(new Parameter("x", parseTypeSignature(INTEGER))),
+            parseTypeSignature(INTEGER),
+            "square",
+            RoutineCharacteristics.builder()
+                    .setDeterminism(DETERMINISTIC)
+                    .setNullCallClause(RETURNS_NULL_ON_NULL_INPUT)
+                    .build(),
+            "RETURN x * x",
+            Optional.empty());
+
+    private RuleTester tester;
+
+    @BeforeTest
+    public void setup()
+    {
+        RuleTester tester = new RuleTester();
+        FunctionManager functionManager = tester.getMetadata().getFunctionManager();
+        InMemoryFunctionNamespaceManager namespaceManager = new InMemoryFunctionNamespaceManager(
+                "unittest",
+                new SqlInvokedFunctionNamespaceManagerConfig().setSupportedFunctionLanguages("{\"sql\": \"SQL\",\"java\": \"THRIFT\"}"));
```

Comment:
You don't have to call `setSupportedFunctionLanguages` here, the default (SQL only) should work. But if you want to add THRIFT function, might as well add a test to make sure that won't be inlined. 😂  You can also add tests to make sure builtin SQL functions like `array_sum` is also inlined correctly if you want.

Reply:
Added tests for both cases